server/src/controllers/orderController.service.ts

```diff
@@ -34,23 +37,62 @@ export const ticketingWebhook = async (
       break;
     }
     case 'checkout.session.expired':
-      await updateCanceledOrder(prisma, order);
+      await updateCanceledOrder(prisma, order, false);
       break;
   }
 };
 
+export const readerWebhook = async (
+  prisma: ExtendedPrismaClient,
+  eventType: string,
+  errMsg: string,
+  paymentIntent: string,
+) => {
+  const socketURL = process.env.WEBSOCKET_URL;
+  if (socketURL === undefined) {
+    console.error('websocket url undefined');
+    return;
+  }
+  const ws = new WebSocket(socketURL);
+  ws.on('error', console.error);
+  ws.on('open', () => {
+    const messageType = 'reader';
+    ws.send(JSON.stringify({messageType, paymentIntent, eventType, errMsg}));
+    ws.close();
+  });
+
+  switch (eventType) {
+    case 'payment_intent.payment_failed':
+      const order = await prisma.orders.findFirst({
+        where: {
+          payment_intent: paymentIntent,
+        },
+      });
+      if (!order) return;
+      console.log(order);
```

Comment:
We typically don't need console.logs in the server

Reply:
removed all console logs from server